File: Spring backend/src/main/java/br/edu/ifsp/finances/domain/request/TransactionRequest.java
package br.edu.ifsp.finances.domain.request;

import lombok.Data;

import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Positive;
import java.math.BigDecimal;

@Data
public class TransactionRequest {

    @NotBlank
    private String name;

    @Positive
    private BigDecimal amount;

    @NotNull
    @Positive
    private Long account;

    @NotNull
    @Positive
    private Long category;

}

File: Android app/app/src/main/java/br/edu/ifsp/finances/domain/request/TransactionRequest.kt
package br.edu.ifsp.finances.domain.request

import com.google.gson.annotations.SerializedName
import java.math.BigDecimal

data class TransactionRequest (
    @SerializedName("name")
    val name : String,
    @SerializedName("amount")
    val amount : BigDecimal,
    @SerializedName("account")
    val account : Long,
    @SerializedName("category")
    val category : Long
)